src/main/java/eiteam/esteemedinnovation/commons/Config.java

```diff
@@ -58,400 +44,48 @@
-        enableThaumcraftIntegration = config.get("Integration", "Enable Thaumcraft", true).getBoolean();
-        enableNitorPoweredCrucible = config.get("Integration", "Allow the Thaumcraft Nitor to power the Crucible", true).getBoolean();
-        enableBotaniaIntegration = config.get("Integration", "Enable Botania", true).getBoolean();
-        enableEnchiridionIntegration = config.get("Integration", "Enable Enchiridion", true).getBoolean();
-        enableTwilightForestIntegration = config.get("Integration", "Enable Twilight Forest", true).getBoolean();
-        enableBloodMagicIntegration = config.get("Integration", "Enable Blood Magic", true).getBoolean();
-        enableEnderIOIntegration = config.get("Integration", "Enable Ender IO", true).getBoolean();
-        enableThermalFoundationIntegration = config.get("Integration", "Enable Thermal Foundation", true).getBoolean();
-        enableIC2Integration = config.get("Integration", "Enable IC2", true).getBoolean();
-        enableNaturaIntegration = config.get("Integration", "Enable Natura", true).getBoolean();
-        enableTinkersConstruct = config.get("Integration", "Enable Tinker's Construct", true).getBoolean();
-        enableRailcraftIntegration = config.get("Integration", "Enable Railcraft", true).getBoolean();
-        enableNEIIntegration = config.get("Integration", "Enable NEI", true).getBoolean();
-
-        hasAllCrucial = enableBoiler && enableGauge && enableTank && enablePipe;
+        enableThaumcraftIntegration = config.get(CATEGORY_INTEGRATION, "Enable Thaumcraft", true).getBoolean();
+        enableNitorPoweredCrucible = config.get(CATEGORY_INTEGRATION, "Allow the Thaumcraft Nitor to power the Crucible", true).getBoolean();
+        enableBotaniaIntegration = config.get(CATEGORY_INTEGRATION, "Enable Botania", true).getBoolean();
+        enableEnchiridionIntegration = config.get(CATEGORY_INTEGRATION, "Enable Enchiridion", true).getBoolean();
+        enableTwilightForestIntegration = config.get(CATEGORY_INTEGRATION, "Enable Twilight Forest", true).getBoolean();
+        enableBloodMagicIntegration = config.get(CATEGORY_INTEGRATION, "Enable Blood Magic", true).getBoolean();
+        enableEnderIOIntegration = config.get(CATEGORY_INTEGRATION, "Enable Ender IO", true).getBoolean();
+        enableThermalFoundationIntegration = config.get(CATEGORY_INTEGRATION, "Enable Thermal Foundation", true).getBoolean();
+        enableIC2Integration = config.get(CATEGORY_INTEGRATION, "Enable IC2", true).getBoolean();
+        enableNaturaIntegration = config.get(CATEGORY_INTEGRATION, "Enable Natura", true).getBoolean();
+        enableTinkersConstruct = config.get(CATEGORY_INTEGRATION, "Enable Tinker's Construct", true).getBoolean();
+        enableRailcraftIntegration = config.get(CATEGORY_INTEGRATION, "Enable Railcraft", true).getBoolean();
+        enableNEIIntegration = config.get(CATEGORY_INTEGRATION, "Enable NEI", true).getBoolean();
+
+        // TODO: Abstract this somehow
```

Comment:
@ExpensiveKoala Do you have any ideas for this? The one thing that stands out in my mind would be something like

```java
hasAllCrucial = true;
for (ConfigurableModule module : ContentModuleHandler.configurableModules) {
    if (!module.areCrucialOptionsEnabled()) {
        hasAllCrucial = false;
        break;
    }
}

// and in ConfigurableModule
default boolean areCrucialOptionsEnabled() {
    return true;
}

// example implementation in BoilerModule
@Override
public void areCrucialOptionsEnabled() {
    return enableBoiler;
}
```

Reply:
I suppose that would be better than what is currently there.